### ISSUE TITLE
Training with own dataset, some doubts. 

### ISSUE THREAD

**Issue by astronaut71** [AlexeyAB/darknet#2274]
Hi 

Im new to darknet and CNN. I would like to detect my own objects, and for start only persons (workers in the construction field).  So I have only one class: workers. So.just for start and get dome feeling  I labelled 750 images with workers on the construction field (in most of the images there are 5 or even more workers). Then I follow the link [https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects]  . Then I start training by using the command line: `./darknet detector train /darknet/data/workers.data /darknet/cfg/yolo-obj.cfg darknet19_448.conv.23`

But Im running this one day already and is still processing the traing. At the moment have this on my terminal : 50: 595.310791, 600.561157 avg, 0.000000 rate, 1161.429639 seconds, 3200 images
Resizing
480

Why 3200 images when I have only 750 training images? Any help what is going one? I enclose the screenshot
![screenshot from 2019-01-23 11-38-59](https://user-images.githubusercontent.com/2849083/51583266-b8148f80-1f0a-11e9-8306-6a21a905675a.png)
 of the terminal

**Comment by PabloDIGITS** [comment]
I think your training is going well, you have to keep going until your avg loss gets lower than 1.

You have 3072 images because you have done 48 iterations of 64 images each one: 48*64=3072. The same image is going into the network several times. That is normal, in my opinion.

**Comment by astronaut71** [comment]
Ok. So because of  the yolo-obj.cfg the line batch=64 I have 64 images? How this 64 images are chosen? And why exactly 64? Sorry for such questions but im very new to this

**Comment by PabloDIGITS** [comment]
> So because of the yolo-obj.cfg the line batch=64 I have 64 images?
Yes thats the reason.

> How this 64 images are chosen? 
I think they are chosen randomly but I am not sure at 100%. @AlexeyAB can correct me if I´m wrong.

> why exactly 64?
I dont why exactly this number. But it means that the network uses 64 images to perform the backpropagation algorithm. Maybe this number is product of some trade-off between accuracy of the network and computational operations.

**Comment by astronaut71** [comment]
ok. I see. Playing with this number does it affect the accuracy ?

**Comment by astronaut71** [comment]
So when have this in terminal 
48: 584.801147, 601.866760 avg, 0.000000 rate, 1146.490854 seconds, 3072 images
Loaded: 0.000041 seconds
Means iteration =48,  avg loss =601.87 , rate =0? what is the rate?

and then in iteration 54: 1065.992554, 763.783630 avg, 0.000000 rate, 2304.178075 seconds, 3456 images

avg loss =763? is getting bigger. Is that normal?

**Comment by PabloDIGITS** [comment]
If it changes the number of images used for the backpropagation algorithm I would say yes.
You can see the avg loss in the image you attached in the 48th iteration is 601.866. You should see a graph of the loss with the command that you use.

> what is the rate?

I dont know what is the rate parameter yet.

> avg loss =763? is getting bigger. Is that normal?

It can be normal if it gets bigger for a few itertions and then go back to descendent. But if only is raising there are something wrong with your training.

**Comment by PabloDIGITS** [comment]
I think your number of filters in yolo layers is wrong. If you just want to detect one class the number should be: (num_classes + 5)*3 = 18.

**Comment by astronaut71** [comment]
sorry, I made mistake when put the question. Yes, number of filters is 18. So in lines 603, 689 and 776 of the yolo-obj.cfg file the line is filters=18. in the other lines I didnt change the number of filters. is that the correct way? Why im getting such big number of loss?

**Comment by astronaut71** [comment]
Cause there are 75 lines where filters appears. Should I change all those 75 lines in filters=18 or only lines 603, 689 and 776 of the yolo-obj.cfg file?

**Comment by PabloDIGITS** [comment]
Only change the number of filters of the convolutional layer before the yolo layer. The line is not the same for me, I´m using yolov3-tiny.cfg

**Comment by astronaut71** [comment]
yes, I have done that. So by 50 iterations is normal to have avg loss of 700?

**Comment by PabloDIGITS** [comment]
I think is normal. Keep training and see if it decreases.

**Comment by astronaut71** [comment]
ok. thanks

**Comment by astronaut71** [comment]
I update the question, so now the filters number is correct 18.

**Comment by AlexeyAB** [comment]
@astronaut71 Hi,

Do you have any nVidia GPU?
Without it you will train the network all your life.

**Comment by astronaut71** [comment]
Yes I have nVidia GTX 1060 GPU. Im using Ubuntu 16.04. Which version of CUDA is need it? Can it work with CUDA8?

    On Wednesday, January 23, 2019, 11:10:36 PM GMT+8, Alexey <notifications@github.com> wrote:  
 
 
@astronaut71 Hi,

Do you have any nVidia GPU?
Without it you will train the network all your life.

**Comment by AlexeyAB** [comment]
@astronaut71 

Yes, it should work with CUDA 8.0. And use `Download cuDNN v7.1.4 (May 16, 2018), for CUDA 8.0`: https://developer.nvidia.com/rdp/cudnn-archive

Then set `GPU=1 CUDNN=1 OPENCV=1` in the `Makefile`  https://github.com/AlexeyAB/darknet#how-to-compile-on-linux

> * GPU=1 to build with CUDA to accelerate by using GPU (CUDA should be in /usr/local/cuda)
> * CUDNN=1 to build with cuDNN v5-v7 to accelerate training by using GPU (cuDNN should be in /usr/local/cudnn)
> * OPENCV=1 to build with OpenCV 3.x/2.4.x - allows to detect on video files and video streams from network cameras or web-cams

and do
```
make clean
make
```

----

> Why 3072 images when I have only 750 training images? 

Due to automatic data augmentation in Yolo - it randomly loads your images and changes them.

**Comment by astronaut71** [comment]
ok. Thanks. And how about Cuda 9.0?

**Comment by astronaut71** [comment]
Also which cuDNN  should I install? cuDNN Runtime Library for Ubuntu16.04 (Deb) or cuDNN Developer Library for Ubuntu16.04 (Deb)? Sorry, for so many questions but Im new to this framework

**Comment by AlexeyAB** [comment]
@astronaut71 
 cuDNN Developer Library for Ubuntu16.04 (Deb)

https://github.com/AlexeyAB/darknet#you-only-look-once-unified-real-time-object-detection-versions-2--3

> This repository supports:
> 
> both Windows and Linux
> both OpenCV 2.x.x and OpenCV <= 3.4.0 (3.4.1 and higher isn't supported, but you can try)
> both cuDNN >= v7
> CUDA >= 7.5
> also create SO-library on Linux and DLL-library on Windows
